test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java

```diff
@@ -47,6 +49,22 @@ public NioSocketChannel(String profile, SocketChannel socketChannel, SocketSelec
     }
 
     @Override
+    public void sendMessage(BytesReference reference, ActionListener<TcpChannel> listener) {
+        // TODO: temporar
+        writeContext.sendMessage(reference, new ActionListener<NioChannel>() {
```

Comment:
can't you just use the passed in listener?

Reply:
No. ActionListener<TcpChannel> != ActionListener<NioChannel> to the compiler. I will fix this in a follow up (https://github.com/elastic/elasticsearch/pull/27407#issuecomment-344734284).